extensions/offline-snooty/src/convertGatsbyToHtml/fileHandler.ts

```diff
@@ -0,0 +1,47 @@
+import { HTMLAnchorElement, HTMLImageElement, Node, Window } from "happy-dom";
+import { promises as fsPromises } from "node:fs";
+
+function updateToRelativePaths(nodeList: Node[], prefix: string) {
+  // for links: href = relativePath + href + index.html
+  // for images: src = relativePath + src
+  for (let index = 0; index < nodeList.length; index++) {
+    const node = nodeList[index];
+    if (node instanceof HTMLAnchorElement) {
+      if (!node["href"].startsWith("/")) {
+        continue;
+      }
+      node["href"] = (prefix + node["href"] + "/index.html").replace(
+        /\/+/,
+        "/"
+      );
+    } else if (node instanceof HTMLImageElement) {
+      if (!node["src"].startsWith("/")) {
+        continue;
+      }
+      node["src"] = (prefix + node["src"]).replace(/\/+/, "/");
+    }
```

Comment:
Nit: I think it would be cleaner to refactor so it looks like so
```suggestion
     if (node["href"].startsWith("/")) {
        if (node instanceof HTMLAnchorElement) {
             node["href"] = (prefix + node["href"] + "/index.html").replace(
        /\/+/,
        "/"
      );
      }
       else if (node instanceof HTMLImageElement) {
           node["src"] = (prefix + node["src"]).replace(/\/+/, "/");
    }
```

Reply:
it is a bit messy here, but `node['href']` can only be used after verifying it is an anchor element due to happy dom's typing
`  Property 'href' does not exist on type 'Node'.ts(7053)`

Reply:
What would you think about simplifying so there are only two conditionals that look like the following? Would this work?
`if (node instanceof HTMLImageElement && node["src"].startsWith("/"))` and 
`if (node instanceof HTMLAnchorElement && node["href"].startsWith("/"))`

Reply:
good point! done!